itests/eth_fee_history_test.go

```diff
@@ -17,6 +19,21 @@ import (
 	"github.com/filecoin-project/lotus/node/impl/full"
 )
 
+func getAnswer(tsHeights []int, requestAmount, startHeight int) (count, oldestHeight int) {
+	idx := sort.SearchInts(tsHeights, startHeight)
+	// handle null rounds
+	for tsHeights[idx] > startHeight {
+		idx--
+	}
+	cnt := requestAmount
+	oldestIdx := idx - requestAmount + 1
+	if idx < cnt {
+		cnt = idx + 1
```

Comment:
Not sure I follow this.

If you have `tsHeights = [1, 2, 3, 4, 6, 8]`, and you request `startHeight = 5`, you will have `idx = 4` (current value 6). If you request 10 heights, you will enter this if, and will end up returning cnt = 5, even though you would only process 4 heights (1..4)?

Reply:
In your example, idx would be 3 (idx would be the index of the largest number that's smaller than the target), so it would return 4 entries. I can rewrite it to make it clearer.

Reply:
Hm, not according to the Godocs and this playground: https://play.golang.com/p/bQyjTqj9Gxb

Reply:
There's a [for loop](https://github.com/filecoin-project/lotus/pull/10413/files/1747893ead537a24b39082c2f7000ceb742b5e13..61d2d461e1e4dc29b360f095d070b48377922c7d#diff-4ec74518af0f29b47821833fc7b7ff99d23cb2b21ce3bfe7deeb6c9994f64646R25-R27) that decreases the idx